compiler/rustc_mir/src/transform/inline.rs

```diff
@@ -48,6 +49,7 @@ impl<'tcx> MirPass<'tcx> for Inline {
                     tcx,
                     param_env: tcx.param_env_reveal_all_normalized(body.source.def_id()),
                     codegen_fn_attrs: tcx.codegen_fn_attrs(body.source.def_id()),
+                    inlined: Default::default(),
```

Comment:
Would it make sense to use something like `RefCell` here? To a caller, `run_pass` doesn't really seem like its mutating the struct right? I am not sure if the runtime checks costly though.

Reply:
`Inliner` is an implementation detail. It is created and used in a single place, so its API is pretty much insignificant. The `MirPass impl for Inline` is a more public counterpart which uses `&self` as a receiver already.